source/fab/steps/grab/git.py

```diff
@@ -3,113 +3,98 @@
 
 from fab.steps.grab import GrabSourceBase
+from fab.tools import run_command
 
-try:
-    import git
-except ImportError:
-    git = None  # type: ignore
 
+def current_commit(folder=None):
+    folder = folder or '.'
+    output = run_command(['git', 'log', '--oneline', '-n', '1'], cwd=folder)
+    commit = output.split()[0]
+    return commit
 
-class GrabGit(GrabSourceBase):
-    """
-    Grab a Git repo into the project workspace.
-
-    A destination name must be specified because each git repo needs to be in a separate folder.
-
-    .. note::
-
-        A revision must be specified.
-        If `shallow` is set (the default), then the revision should be a branch or tag.
-        If `shallow` is not set, the revision can also be a commit hash.
-
-    A `shallow` grab clones/fetches only the given revision, with no history.
-    Otherwise, the full repo is cloned including all branches and history.
 
-    Example::
-
-        GrabGit(src='~/git/my_project', revision='v0.1b')
-        GrabGit(src='https://github.com/me/my_project.git', revision='a1b2c3', shallow=False)
+class GrabGitBase(GrabSourceBase, ABC):
+    """
+    Base class for Git operations.
 
     """
+    def run(self, artefact_store: Dict, config):
+        if not self.tool_available():
+            raise RuntimeError("git command line tool not available")
+        super().run(artefact_store, config)
 
-    def __init__(self, src: Union[Path, str], dst: str = None,  # type: ignore
-                 revision=None, shallow: bool = True, name=None):
-        """
-        Params as for :class:`~fab.steps.grab.GrabSourceBase`, plus the following.
+    def tool_available(self) -> bool:
+        """Is the command line git tool available?"""
+        try:
+            run_command(['git', 'help'])
+        except FileNotFoundError:
+            return False
+        return True
 
-        :param shallow:
-            This flag causes the grab to be quick and shallow, fetching just the branch and no history.
-            You may need to turn this off when fetching a commit hash.
+    def is_working_copy(self, dst: Union[str, Path]) -> bool:
+        """Is the given path is a working copy?"""
+        try:
+            run_command(['git', 'status'], cwd=dst)
+        except RuntimeError:
+            return False
+        return True
 
-        """
-        if not revision:
-            raise ValueError("GrabGit (currently) requires a revision to be specifed.")
+    def fetch(self):
+        # todo: allow shallow fetch with --depth 1
+        command = ['git', 'fetch', self.src]
+        if self.revision:
+            command.append(self.revision)
 
-        if not dst:
-            raise ValueError("A destination name must be specified to GrabGit.")
+        run_command(command, cwd=str(self._dst))
 
-        super().__init__(src=str(src), dst=dst, revision=revision, name=name)
 
-        self.shallow = shallow
+# todo: allow cli args, e.g to set the depth
```

Comment:
Is there an issue capturing this todo?

Reply:
[Created one](https://github.com/orgs/metomi/projects/1/views/1?filterQuery=git&pane=issue&itemId=22270115), thank you.